components/dashboard/src/settings/Plans.tsx

```diff
@@ -112,15 +79,35 @@ export default function () {
     const paidPlan = paidSubscription && Plans.getById(paidSubscription.planId);
 
     const assignedTeamSubscriptions = activeSubscriptions.filter(s => AssignedTeamSubscription.is(s));
-    console.log('assignedTeamSubscriptions', assignedTeamSubscriptions);
     const getAssignedTs = (type: PlanType) => assignedTeamSubscriptions.find(s => {
         const p = Plans.getById(s.planId);
         return !!p && p.type === type
     });
-    const assignedUnleashedTs = getAssignedTs('professional');
     const assignedProfessionalTs = getAssignedTs('professional-new');
+    const assignedUnleashedTs = getAssignedTs('professional');
     const assignedStudentUnleashedTs = getAssignedTs('student');
-    const assignedTs = assignedUnleashedTs || assignedProfessionalTs || assignedStudentUnleashedTs;
+    const assignedTs = assignedProfessionalTs || assignedUnleashedTs || assignedStudentUnleashedTs;
+
+    const claimedTeamSubscriptionId = new URL(window.location.href).searchParams.get('teamid');
+    if (!!claimedTeamSubscriptionId && !assignedTeamSubscriptions.some(s => !!s.teamSubscriptionSlotId)) {
```

Comment:
it's now unclear, when this is actually executed 🤔 
could you explain the reason?

before, the `tsGetUnassignedSlot` was called on mount, right?

Reply:
This simplifies the code a bit (no need for an inline function that is just called once).

Also it's now called on `render()`, when there is a `teamid` in the URL query parameters. Also, as soon as there is a response, the query parameters are stripped away from the URL before showing the modal, so that visiting a team invite links effectively only shows the modal once (and the user can then accept or dismiss it).

Reply:
the code itself looks cleaner indeed.

it still looks like `tsGetUnassignedSlot` may be called more than once. see, the only thing which would stop it from being recalled is the history patching, which happens after `tsGetUnassignedSlot` returns. in the meantime there is a chance that other state manipulations will cause a render cycle.

> effectively only shows the modal once
you are right about that.

Reply:
> it still looks like `tsGetUnassignedSlot` may be called more than once.

You're right. I assumed this wasn't a problem (because it doesn't modify anything), but I agree it's probably cleaner to call it just once.

Reply:
Fixed by implementing your suggestion: https://github.com/gitpod-io/gitpod/pull/3873#discussion_r615009593